core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -604,8 +621,37 @@ private void catchUp(JournalStateMachine stateMachine, LocalFirstRaftClient clie
       } catch (TimeoutException | ExecutionException | IOException e) {
         ex = e;
       }
+
       if (ex != null) {
-        LOG.info("Exception submitting term start entry: {}", ex.toString());
+        // LeaderNotReadyException typically indicates Ratis is still replaying the journal.
+        long now = System.currentTimeMillis();
+        // powers of 2, but capped to the JOURNAL_STAT_MAX_LOG_INTERVAL_MS
+        long nextLogTime = 1000L * Math.min(
+            1L << (logCount > 30 ? 30 : logCount),
+            JOURNAL_STAT_LOG_MAX_INTERVAL_MS);
+        if (ex instanceof LeaderNotReadyException
+            && (now - lastMeasuredTime) > nextLogTime) {
+          logCount++;
+          long currCommitIdx = stateMachine.getLastAppliedCommitIndex();
+          long timeSinceLastMeasure = (now - lastMeasuredTime);
+          long commitIdxRead = currCommitIdx - lastCommitIdx;
+          double commitIdxRateS = 1000 * ((double) commitIdxRead) / timeSinceLastMeasure;
+          StringJoiner logMsg = new StringJoiner("|");
+          logMsg.add(String.format("current SN: %d", currCommitIdx));
+          logMsg.add(String.format("entries in last %dms=%d", timeSinceLastMeasure, commitIdxRead));
+          if (endCommitIndex.isPresent()) {
+            long commitsRemaining = endCommitIndex.getAsLong()
+                - stateMachine.getLastAppliedCommitIndex();
```

Comment:
Would this be negative after we submit a new catchup entry?

Reply:
The catchup entry value != commit index. The commit indexes come from ratis and as far as I can tell are not negative

Reply:
Sorry about the confusion. I was thinking about the whole expression `endCommitIndex.getAsLong() - stateMachine.getLastAppliedCommitIndex()` being negative due to the LastAppliedCommitIndex move beyond initial endCommitIndex with multiple catchup entries submitted in the while loop.

Reply:
Oh, I see what you're saying. Yes, it's potentially negative if the journal replay takes that long.

How long do we have to replay until that really matters..well..let's do some math.

Max logging interval = 30sec
Journal replay rate = 20k/s (this is approximate from my local testing. I hit 28k, but let's say it's slower on a real cluster)

catchup entries submitted before we log a negative "commits remaining" (20k/s * 30s) = 600,000 entries
Say we submit 1 catchup entry/second (current default afaik)
Then it would take 600,000 seconds of journal replay before we logged a negative value one time. This translates to 166.66 hours of replay or about 6.9 days of  the master simply replaying the journal to start up. 

Frankly, I don't think its a huge issue if this is negative. I can add in some defensive statements just in case to prevent logging negative values. But generally the amount of time we would need to replay for this becoming an issue is somewhat ridiculous and I hope no one would ever let their journal get that large before taking a checkpoint.

Reply:
updated